examples/08_inplace_FFT/08_inplace_FFT.cpp

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: (C) The kokkos-fft development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0 WITH LLVM-exception
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Complex.hpp>
+#include <Kokkos_Random.hpp>
+#include <KokkosFFT.hpp>
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+template <typename T>
+using RightView2D = Kokkos::View<T **, Kokkos::LayoutRight, execution_space>;
+
+template <std::size_t DIM>
+using axis_type = KokkosFFT::axis_type<DIM>;
+template <std::size_t DIM>
+using shape_type = KokkosFFT::shape_type<DIM>;
+
+int main(int argc, char *argv[]) {
+  Kokkos::initialize(argc, argv);
+  {
+    constexpr int n0 = 128, n1 = 128;
+    const Kokkos::complex<double> z(1.0, 1.0);
+
+    shape_type<2> shape;
+    shape[0] = n0;
+    shape[1] = n1;
+
+    // Forward and backward complex to complex transform
+    // Define a 2D complex view to handle data
+    RightView2D<Kokkos::complex<double>> xc2c("xc2c", n0, n1);
+
+    // Create an unmanaged view on the same data
+    RightView2D<Kokkos::complex<double>> xc2c_hat(xc2c.data(), n0, n1),
+        xc2c_inv(xc2c.data(), n0, n1);
+
+    // Fill the input view with random data
+    Kokkos::Random_XorShift64_Pool<> random_pool(12345);
+    execution_space exec;
+    Kokkos::fill_random(exec, xc2c, random_pool, z);
+
+    KokkosFFT::fft2(exec, xc2c, xc2c_hat);
+    KokkosFFT::ifft2(exec, xc2c_hat, xc2c_inv);
+
+    // Real to complex transform
+    // Define a 2D complex view to handle data
+    RightView2D<Kokkos::complex<double>> xr2c_hat("xr2c", n0, n1 / 2 + 1);
+
+    // Create an unmanaged view on the same data with the FFT shape,
+    // that is (n0, n1) -> (n0, n1/2+1) R2C transform
+    // The shape is incorrect from the view point of casting to real
+    // For casting, the shape should be (n0, (n0/2+1) * 2)
+    RightView2D<double> xr2c(reinterpret_cast<double *>(xr2c_hat.data()), n0,
+                             n1),
```

Comment:
Should we provide helper functions to avoid users using `reinterpret_cast` ?

Reply:
That may be ideal. 
Can I make it another PR?